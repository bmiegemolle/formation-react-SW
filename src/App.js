import React from 'react';
import {connect} from 'react-redux';
import './App.css';
import Task from './Task';
import {updateTasks} from './actions/todolist';
import _ from 'lodash';

export class App extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      newTask: ''
    };    
  }

  handleToggle = (task, index) => {
    const newTasks = _.cloneDeep(this.props.tasks);
    newTasks[index].done = !task.done;
    this.props.updateTasks(newTasks);
  }

  handleNewTaskChange = (event) => {
    this.setState({newTask: event.target.value});
  }

  handleNewTaskSubmit = (event) => {
    event.preventDefault();
    this.props.updateTasks([...this.props.tasks, {
      name: this.state.newTask, 
      done: false}
    ]);
    this.setState({
      newTask: ''
    });
  }

  render() {
    return (
      <div className="App">
        {this.props.tasks.map((task, index) => (
            <Task key={index} name={task.name} done={task.done} handleToggle={this.handleToggle.bind(this, task, index)}
            />
        ))}
        <form onSubmit={this.handleNewTaskSubmit}>
          <label>
            New task:
            <input type="text" value={this.state.newTask} onChange={this.handleNewTaskChange} />
          </label>
          <input type="submit" value="Add" />
        </form>

      </div>
    );
  }
}

const mapStateToProps = (state)=>{
  return{
    tasks: state.tasks,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    updateTasks: (tasks) => dispatch(updateTasks(tasks)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
