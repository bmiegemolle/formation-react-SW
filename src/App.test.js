import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {App} from './App';

describe("<App />", () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App tasks={[]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<App tasks={[{name:'Task1', done:true}, {name:'Task2', done:false}]}/>).toJSON();
    expect(wrapper).toMatchSnapshot()
  })

});