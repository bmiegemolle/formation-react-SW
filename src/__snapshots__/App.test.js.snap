// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Task-done"
    onClick={[Function]}
  >
    Task1
  </div>
  <div
    className="Task-todo"
    onClick={[Function]}
  >
    Task2
  </div>
  <form
    onSubmit={[Function]}
  >
    <label>
      New task:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Add"
    />
  </form>
</div>
`;
